src/CTA.Rules.Common/CsprojManagement/CsprojXDocument.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System.Linq;
+using System.Text.RegularExpressions;
+using System.Xml.Linq;
+using CTA.Rules.Common.Extensions;
+
+namespace CTA.Rules.Common.CsprojManagement
+{
+    public class CsprojXDocument
+    {
+        private readonly XDocument _csproj;
+
+        private string _targetFrameworksValue;
+        private string TargetFrameworksValue
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_targetFrameworksValue))
+                {
+                    _targetFrameworksValue = GetTargetFrameworksValue();
+                }
+
+                return _targetFrameworksValue;
+            }
+        }
+
+        public CsprojXDocument(XDocument csproj)
+        {
+            _csproj = csproj ?? new XDocument();
+        }
+
+        public bool IsDotnetFramework()
+        {
+            return IsRegexMatch(Constants.DotnetFrameworkPattern, TargetFrameworksValue)
+                   || IsRegexMatch(Constants.DotnetFrameworkSdkPattern, TargetFrameworksValue);
+        }
+
+        public bool IsDotnetCore()
+        {
+            return IsRegexMatch(Constants.DotnetCoreAppPattern, TargetFrameworksValue)
+                   || IsRegexMatch(Constants.DotnetCorePattern, TargetFrameworksValue);
+        }
+
+        public bool IsDotnetStandard()
+        {
+            return IsRegexMatch(Constants.DotnetStandardPattern, TargetFrameworksValue);
+        }
+
+        private string GetTargetFrameworksValue()
+        {
+            var targetFrameworks =
+                _csproj.GetElementsByLocalName(Constants.TargetFrameworkVersionElement).FirstOrDefault()
+                ?? _csproj.GetElementsByLocalName(Constants.TargetFrameworkElement).FirstOrDefault()
+                ?? _csproj.GetElementsByLocalName(Constants.TargetFrameworksElement).FirstOrDefault();
```

Comment:
Can both TargetFramework and TargetFrameworks exist at the same time? If yes, this will not give a correct result

Reply:
Yes, it is technically valid for both `TargetFramework` and `TargetFrameworks` to exist in a .csproj file, but in those cases, `TargetFrameworks` will be ignored ([source](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#targetframeworks:~:text=This%20property%20is%20ignored%20if%20TargetFramework%20(singular)%20is%20specified.))